pandas/core/frame.py

```diff
@@ -2641,7 +2641,7 @@ def reindexer(value):
             from pandas.core.series import _sanitize_index
 
             # turn me into an ndarray
-            value = _sanitize_index(value, self.index, copy=False)
+            value = _sanitize_index(value, self.index, copy=True)
             if not isinstance(value, (np.ndarray, Index)):
                 if isinstance(value, list) and len(value) > 0:
```

Comment:
e.g. I think for ``object`` dtype you *always* need to copy, but for others you may not.